.github/workflows/py-unittests.yml

```diff
@@ -63,28 +70,30 @@ jobs:
       - name: Test with pytest (builder)
         run: |
           PYTHONPATH=. coverage run --parallel-mode -m pytest -v -rP ./tools/cellxgene_census_builder/tests/
-      - uses: actions/upload-artifact@v3
+      - uses: actions/upload-artifact@v4
         with:
-          name: coverage
+          name: coverage-builder-${{ matrix.os }}-${{ matrix.python-version }}
           path: ./.coverage*
           retention-days: 3
 
   submit-codecoverage:
     needs:
       - unit_tests_builder
       - unit_tests_python_api
-    runs-on: ubuntu-20.04
+    runs-on: [self-hosted, Linux]
```

Comment:
this one is okay to run on arm or x86 right? otherwise, we should pick one architecture

Reply:
this step is pure python, so it doesn't care....

Reply:
I spoke too soon - it failed!

The version '3.10' with architecture 'arm64' was not found for Ubuntu 20.04. The list of all available versions can be found here: https://raw.githubusercontent.com/actions/python-versions/main/versions-manifest.json

will add the x86 tag....

Reply:
It is wild how many things still don't work on ARM, even with languages like python. I've had to do similar things for a few of our github actions